### ISSUE TITLE
[proc] Improved Clarity being consumed on hit.

### ISSUE THREAD

**Issue by Bejkon** [WarEmu/WarBugs#19810]
Hej,

**TL;DR:
Improved Clarity currently consumes the buff on hit, when it should be on ability use.**

Improved Clarity is a proc present as a setbonus on a few classes(in this case tested on SW). It grants a buff that should make the next ability free to cast among some other boni. Currently however, this buff gets "spent" by the next hit landed, not the next ability usage.

I started out with testing this in melee(instantcast/DD/non-flight-time) and noticed that the buff got consumed by AAs, making the APcost reduction pointless.

To bypass the AA issue i backed up abit and used a ranged ability(SFA) instead, remaining in assault stance to block any ranged AA from happening. I then noticed that whenever the proc happened and i got the buff, it got consumed by the same ability that had procced it, when it hit the target. This case, again, makes the AP-cost reduction part impossible to use.

I then tried to get the buff to proc with non-damaging abilities, which worked, and then use a damaging ability, just to see if the buff worked at all. By doing this i noticed that the buff would reduce the cost of all abilities used, without being consumed, for all of its 10s uptime, as long as i didnt hit anything.

Here is the TT of Improved Clarity:
![Improved Clarity](https://user-images.githubusercontent.com/35146895/153724499-760312aa-b5ad-45d8-8e1a-5afa9b059c2f.JPG)

I ran the test on Clarity aswell, and it works just as expected from reading the tooltip, Procs on ability use, gets spent on the next ability use. No issue found there.

Here is the TT for Clarity:
![Clarity](https://user-images.githubusercontent.com/35146895/153724559-fb57ad9f-ceb9-48f7-9e2d-d7f0f8b8c8e5.JPG)

//Bejk

**Comment by HerkRoR** [comment]
Having tested Improved Clarity with a four pieces triumphant gear on SW, I can confirm the very accurate report by Bejkon.
Besides, any hit crits when Improved Clarity is triggered, being the ability that triggered it or an auto attack that slips through.

**Comment by gats69** [comment]
Hi,
I am currently having the same problem on my SW:
Improved Clarity proc is very strange, it procs when I mount up, when I change stance, but rarely when I use an ability.
I bought this equipment specifically for this proc 4 and I find myself at a loss because it does not work.
Can you fix this bug please?
